### PR TITLE
Fix message publish in transaction

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes the below error log on deleting a shared network.
```java
2024-04-24 18:24:33,831 ERROR [o.a.c.f.m.MessageBusBase] (API-Job-Executor-14:ctx-63de092f job-2044 ctx-d00db98a) (logid:107d5ed5) NO EVENT PUBLISH CAN BE WRAPPED WITHIN DB TRANSAC
TION!
com.cloud.utils.exception.CloudRuntimeException: NO EVENT PUBLISH CAN BE WRAPPED WITHIN DB TRANSACTION!
        at org.apache.cloudstack.framework.messagebus.MessageBusBase.publish(MessageBusBase.java:167)
        at com.cloud.configuration.ConfigurationManagerImpl.deleteVlanAndPublicIpRange(ConfigurationManagerImpl.java:5345)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
        at com.sun.proxy.$Proxy156.deleteVlanAndPublicIpRange(Unknown Source)
        at org.apache.cloudstack.engine.orchestration.NetworkOrchestrator.deleteVlansInNetwork(NetworkOrchestrator.java:3368)

```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 1]
```diff
@@ -3341,8 +3343,14 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
                                 _resourceLimitMgr.decrementResourceCount(networkFinal.getAccountId(), ResourceType.network, networkFinal.getDisplayNetwork());
                             }
                         }
+                        return deletedVlans.second();
```
Comment (sureshanaparti) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:3346]: @vishesh92 seems this return skips publish events below, check it.

Comment (sureshanaparti) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:3346]: ignore - this returns for _Transaction.execute()_

**Review by sureshanaparti — APPROVED** [review]
clgtm

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 2]
```diff
@@ -3341,8 +3343,14 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
...
                     }
                 });
+                if (CollectionUtils.isNotEmpty(deletedVlanRangeToPublish)) {
+                    for (VlanVO vlan : deletedVlanRangeToPublish) {
+                        _messageBus.publish(_name, MESSAGE_DELETE_VLAN_IP_RANGE_EVENT, PublishScope.LOCAL, vlan);
+                    }
+                }
```
Comment (DaanHoogland) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:3353]: new method?

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 3]
```diff
@@ -5962,7 +5960,12 @@ public boolean deleteVlanIpRange(final DeleteVlanIpRangeCmd cmd) {
             throw new InvalidParameterValueException("Please specify a valid IP range id.");
         }
 
-        return deleteVlanAndPublicIpRange(CallContext.current().getCallingUserId(), vlanDbId, CallContext.current().getCallingAccount());
+        VlanVO deletedVlan = deleteVlanAndPublicIpRange(CallContext.current().getCallingUserId(), vlanDbId, CallContext.current().getCallingAccount());
+        if (deletedVlan != null) {
+            messageBus.publish(_name, MESSAGE_DELETE_VLAN_IP_RANGE_EVENT, PublishScope.LOCAL, deletedVlan);
+            return true;
+        }
+        return false;
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:5968]: new method?

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by borisstoyanov** [comment]
@blueorangutan test keepEnv

**Review by borisstoyanov — APPROVED** [review]
LGMT, we currently don't see the stack trace in the logs but there's still the tungsten related error prompted:
```
2024-04-29 12:34:10,969 ERROR [o.a.c.n.t.s.TungstenFabricUtils] (API-Job-Executor-30:ctx-1f6998d8 job-43 ctx-726dda4f) (logid:32e36fb0) No Tungsten-Fabric provider have been found!
2024-04-29 12:34:10,969 ERROR [o.a.c.n.t.s.TungstenServiceImpl] (API-Job-Executor-30:ctx-1f6998d8 job-43 ctx-726dda4f) (logid:32e36fb0) Failed to find a Tungsten-Fabric provider
2024-04-29 12:34:10,977 DEBUG [o.a.c.f.j.i.AsyncJobManagerImpl] (API-Job-Executor-30:ctx-1f6998d8 job-43 ctx-726dda4f) (logid:32e36fb0) Complete async job-43, jobStatus: SUCCEEDED, resultCode: 0, result: org.apache.cloudstack.api.response.SuccessResponse/null/{"success":"true"}
```
I think that is acceptable

**Comment by vishesh92** [comment]
@blueorangutan test

**Pull request merged**